src/accelerate/utils/modeling.py

```diff
@@ -172,10 +172,7 @@ def set_module_tensor_to_device(
         elif value is not None or torch.device(device) != module._parameters[tensor_name].device:
             param_cls = type(module._parameters[tensor_name])
             kwargs = module._parameters[tensor_name].__dict__
-            if param_cls.__name__ == "Int8Params":
-                # downcast to fp16 if any
-                if new_value.dtype == torch.float32:
-                    new_value = new_value.to(torch.float16)
```

Comment:
I think these changes were needed for `Int8Params` for the serialization - can we add a check to add these only for `Int8Params`

Reply:
This looks great to me for backward compatibility!